src/diffusers/models/embeddings.py

```diff
@@ -460,6 +460,22 @@ def forward(self, image_embeds: torch.FloatTensor):
         return image_embeds
 
 
+class MLPProjection(nn.Module):
+    def __init__(self, image_embed_dim=1024, cross_attention_dim=1024):
+        super().__init__()
+
+        self.image_embeds = torch.nn.Sequential(
```

Comment:
Isn't this the same as https://github.com/huggingface/diffusers/blob/636feba552ab1f9a1ecdd3748e1245fa45b76d0d/src/diffusers/models/attention.py#L342

Could we try to re-use this module instead?

Reply:
hi @patrickvonplaten, I cannot use FeedForward because of a circular dependency between attention.py and embeddings.py, but I can write as follows to remove nn.Sequential

```
class MLPProjection(nn.Module):
    def __init__(self, image_embed_dim=1024, cross_attention_dim=1024):
        super().__init__()

        self.net = nn.ModuleList([
            nn.Linear(image_embed_dim, image_embed_dim),
            nn.GELU(),
            nn.Linear(image_embed_dim, cross_attention_dim),
            nn.LayerNorm(cross_attention_dim),
        ])

    def forward(self, image_embeds: torch.FloatTensor):
        for module in self.net:
            image_embeds = module(image_embeds)
        return image_embeds
```

Reply:
hi again, I have found a solution, you were right @patrickvonplaten 

```
class MLPProjection(nn.Module):
    def __init__(self, image_embed_dim=1024, cross_attention_dim=1024):
        super().__init__()
        from .attention import FeedForward
        
        self.ff = FeedForward(image_embed_dim, cross_attention_dim, mult=1, activation_fn="gelu")
        self.norm = nn.LayerNorm(cross_attention_dim)

    def forward(self, image_embeds: torch.FloatTensor):
        return self.norm(self.ff(image_embeds))
```